src/tests/system/tests/test_gpo.py

```diff
@@ -0,0 +1,1056 @@
+"""
+SSSD GPO tests
+
+:requirement: IDM-SSSD-REQ: Group policy host based access control
+
+The following features and parameters are not covered:
+- ad_gpo_cache_timeout
+- ad_gpo_map_network
+- ad_gpo_map_batch
+- ad_gpo_map_service
+- ad_gpo_map_permit
+- ad_gpo_map_deny
+"""
+
+from __future__ import annotations
+
+import pytest
+
+from sssd_test_framework.roles.ad import AD
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.parametrize("method", ["su", "ssh"])
+@pytest.mark.topology(KnownTopology.AD)
+def test_gpo__is_set_to_enforcing_and_users_are_allowed(client: Client, ad: AD, method: str):
+    """
+    :title: Group policy object host base access control is set to enforcing and users are allowed
+    :description:
+        When ad_gpo_access_control is set to enforcing, user access is managed and enforced by the policy.
+        Unlisted users are denied access.
+    :setup:
+        1. Create the following users; 'user1', 'user2'
+        2. Create the following groups; 'group' with member 'user2'
+        3. Create and link the GPO 'site policy' and add 'user1', group and 'Domain Admins' to
```

Comment:
You should put `group` into single-quotes as well.

Why is `Domain Admins` needed? Is it to allow some operations of the framework? If yes, I think a comment about this is needed. Otherwise it is not obvious why it would be needed for the test.

Reply:
Per the Microsoft documentation

```
Allow log on through Remote Desktop Services

This security setting determines which users or groups have permission to log on as a Remote Desktop Services client.

Default:

On workstation and servers: Administrators, Remote Desktop Users.
On domain controllers: Administrators. 

Reply:
I can update it in the framework so it doesn't have to be defined every time.

Reply:
> I can update it in the framework so it doesn't have to be defined every time?

Hi,

I think this would be a good idea. Would it be possible to have some message in the test output whenever the framework adds a user or a group automatically to make debugging easier?

bye,
Sumit

Reply:
Yes, we can do that, but know that all tests *should* contain the user add in the setup and it will fail if the user doesn't exist and will provide the debug output of why the user add did not work. I don't see the value in this now. 

Previously, when the user creation was done outside of the test, yes, now, the test will fail if this step fails.

Reply:
Hi,

sorry, I didn't meant adding users or groups to the system but to the GPOs.

bye,
Sumit